### ISSUE TITLE
Can't install on WSL 2 Windows 10 or Crashes (using faiss-gpu)

### ISSUE THREAD

**Issue by grahama1970** [bclavie/RAGatouille#144]
Ragatouille works great in a colab. I can't seem to get it to install or run in WSL2. I've tried a jupyter notebook and python code as supplied by the docs, to no avail. I'd love to try again when it ready for WSL 2 :) 
I uninstalled faiss-cpu and installed faiss-gpu. I've got a single a5000 GPU 24 gigs.

**Comment by bclavie** [comment]
Hey, thank you for reporting this. I'm curious about the installation issue, do you mean that it doesn't install when running `pip install` (or your package manager of choice)?

**Comment by grahama1970** [comment]
I've tried pip and conda. Can't seem to get either to work for ragatouille, yet. The google colab Jupyter notebook does work.

**Comment by bclavie** [comment]
Could you send over the output you get when you try to run the installation commands?

**Comment by phaistos** [comment]
I have `ragatouille` working under WSL2 with `faiss-gpu`.  Afaik `conda` is the only way to install `faiss-gpu` (and for it to actually work).  I did run into some issues along the way..

If conda installs it's own `libstdc++.so` you may run into an issue where the building of those CPP extensions will fail as your host's compiler may be expecting to link with a different/newer version of the standard library.  You will see a missing GLIBC_... symbol error.   The solution to this problem is to install gcc into your conda environment.

I use pipenv so I had to convince that tool to use the python and site packages from the conda environment to get access to faiss-gpu:

```
conda create python=3.10.11 -n my_env
conda install -n my_env pytorch/label/nightly::faiss-gpu conda-forge::gxx
pipenv --python=$(conda run -n my_env which python) --site-packages install
pipenv run pip uninstall -y faiss-cpu
```

And then I had to make sure the conda compiler was used the first time I run my example:

```
> CXX=~/.conda/envs/my_env/bin/c++ ./ragatouille_example.py
```

Hope this helps, the whole situation is a bit of a mess really.

**Comment by TheMcSebi** [comment]
In case you recently updated/reinstalled cuda on your windows host, you might need to reinstall the cuda-toolkit-12-3 package on wsl. A simple apt update apparently won't do, I recently had to find out.

**Comment by grahama1970** [comment]
When I run the above in WSL2, I get the below on the pipenv line.
pipenv --python=$(conda run -n ragatouille which python) --site-pa
ckages install
Traceback (most recent call last):
  File "/usr/bin/pipenv", line 33, in <module>
    sys.exit(load_entry_point('pipenv==11.9.0', 'console_scripts', 'pipenv')())
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 696, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 621, in make_context
    self.parse_args(ctx, args)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 1018, in parse_args
    rest = Command.parse_args(self, ctx, args)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 875, in parse_args
    parser = self.make_parser(ctx)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 821, in make_parser
    for param in self.get_params(ctx):
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 774, in get_params
    help_option = self.get_help_option(ctx)
  File "/usr/lib/python3/dist-packages/pipenv/cli.py", line 26, in get_help_option
    from .import core
  File "/usr/lib/python3/dist-packages/pipenv/core.py", line 21, in <module>
    import requests
  File "/usr/lib/python3/dist-packages/pipenv/vendor/requests/__init__.py", line 65, in <module>
    from . import utils
  File "/usr/lib/python3/dist-packages/pipenv/vendor/requests/utils.py", line 27, in <module>
    from .cookies import RequestsCookieJar, cookiejar_from_dict
  File "/usr/lib/python3/dist-packages/pipenv/vendor/requests/cookies.py", line 172, in <module>
    class RequestsCookieJar(cookielib.CookieJar, collections.MutableMapping):
AttributeError: module 'collections' has no attribute 'MutableMapping'

**Comment by bclavie** [comment]
As this is a `faiss` issue, it should be fixed in [0.0.8](https://github.com/bclavie/RAGatouille/releases/tag/0.0.8), as long as you are indexing fewer than ~100k documents (with the new tentative faiss replacement just using PyTorch to perform k-means).
